hangman.py

```diff
@@ -0,0 +1,340 @@
+    returns: string (of letters), comprised of letters that represents which letters have not
+      yet been guessed.
+    '''
+
+    end_list = []
+    for letters in list(string.ascii_lowercase):
+        if letters in letters_guessed:
+            continue
+        else:
+            end_list.append(letters)
+
+    str_not_used_letters = ''.join(end_list)
+    return str_not_used_letters
+
+
+def len_secret_word_without_dublicate(secret_word):
+    new_set = set()
+    for symbol in secret_word:
+        new_set.add(symbol)
+    length = len(new_set)
+    return length
+    
+    
+
+def hangman(secret_word, hints_switch = False):
+    '''
+    secret_word: string, the secret word to guess.
+    
+    Starts up an interactive game of Hangman.
+    
+    * At the start of the game, let the user know how many 
+      letters the secret_word contains and how many guesses s/he starts with.
+      
+    * The user should start with 6 guesses
+
+    * Before each round, you should display to the user how many guesses
+      s/he has left and the letters that the user has not yet guessed.
+    
+    * Ask the user to supply one guess per round. Remember to make
+      sure that the user puts in a letter!
+    
+    * The user should receive feedback immediately after each guess 
+      about whether their guess appears in the computer's word.
+
+    * After each guess, you should display to the user the 
+      partially guessed word so far.
+    
+    Follows the other limitations detailed in the problem write-up.
+    '''
+
+    def letters_in_secret_word(secret_word):
+        #Function for getting list of letters from secret word without dublication
+        letters_in_secret_word = []
+        for letters in secret_word:
+            if letters in letters_in_secret_word:
+                continue
+            letters_in_secret_word.append(letters)
+        return letters_in_secret_word
+
+    #Instruction to the game for player
+    letters_in_secret_word = letters_in_secret_word(secret_word)
+    print("Welcome to the game Hangman!")
+    print("I am thinking of a word that is", len(secret_word), "letters long.")
+    print("You have 3 warnings left")
+    guesses_left = guesses_warnings_start[0]
+    warnings_left = guesses_warnings_start[1]
+    letters_guessed = []
+    winning = False
+    while True:
+        print("--------------------------------------")
+        if guesses_left <= 0:
+            #End of the game (losing), winning is still False
+            break
+
+        #Main information for the player
+        print("You have", guesses_left, "guesses left")
+        get_available_letters(letters_guessed)
+        print("Available letters:", get_available_letters(letters_guessed))
+        guessed_letter = input("Please guess a letter: ")
+
+        if hints_switch:
+            #Start game with hints on mode, if hints_switch = True
+            if guessed_letter == "*":
```

Comment:
I'd make "*" symbol as a constant that defined in the top of file

Reply:
done